ghcide/src/Development/IDE/Types/Exports.hs

```diff
@@ -90,26 +97,45 @@ mkIdentInfos mod (AvailTC _ nn flds)
         | n <- nn ++ map flSelector flds
       ]
 
+buildModuleExportMap:: [ModIface] -> Map.HashMap Text [Text]
+buildModuleExportMap modIfaces = do
+  let exports = map extractModuleExports modIfaces
+  Map.fromListWith (<>) exports
+
+extractModuleExports :: ModIface -> (Text, [Text])
+extractModuleExports modIFace = do
+  let modName = pack $ moduleNameString $ moduleName $ mi_module modIFace
+  let ifaces = mi_exports modIFace
+  let functionSet = map (pack . getOccString . availName) ifaces
+  (modName, functionSet)
+
 createExportsMap :: [ModIface] -> ExportsMap
-createExportsMap = ExportsMap . Map.fromListWith (<>) . concatMap doOne
+createExportsMap modIface = do
+  let exportsMap = (Map.fromListWith (<>) . concatMap doOne) modIface
+      moduleExportMap = buildModuleExportMap modIface
+  ExportsMap exportsMap moduleExportMap
   where
-    doOne mi = concatMap (fmap (second Set.fromList) . unpackAvail mn) (mi_exports mi)
-      where
-        mn = moduleName $ mi_module mi
+    doOne mi = do
+      let getModDetails = unpackAvail $ moduleName $ mi_module mi
+      concatMap (fmap (second Set.fromList) . getModDetails) (mi_exports mi)
 
 createExportsMapMg :: [ModGuts] -> ExportsMap
-createExportsMapMg = ExportsMap . Map.fromListWith (<>) . concatMap doOne
+createExportsMapMg modIface = do
+  let exportsMap = (Map.fromListWith (<>) . concatMap doOne) modIface
+  ExportsMap exportsMap Map.empty
```

Comment:
Is it ok to have an empty map for the create methods that do not take modIface? @pepeiborra